### ISSUE TITLE
Autocomplete for import is duplicating module import

### ISSUE THREAD

**Issue by DiegoCoronel** [ceylon/ceylon-ide-eclipse#1602]
On windows environment, using latest master build:

with this:
```
native("jvm")
module person_impl "1.0.0" {
	import java.base "8";
	import person_api "1.0.0";
}
```

when in any ceylon file i try to autocomplete for example: ```import person_api.rest.r``` and with Ctrl+Space then is duplicates the import in module.ceylon.
```
native("jvm")
module person_impl "1.0.0" {
	import java.base "8";
	import person_api "1.0.0";
	import person_api "1.0.0";
}
```

**Comment by bjansen** [comment]
I can't reproduce that :(

Are `person_api` and `person_impl` in the same project? Do you have the same problem if you import things from the SDK?

**Comment by DiegoCoronel** [comment]
@bjansen so, i tried again and Im able to reproduce 100% with these steps:
- In same project create module Foo and module Bar
- In Foo create a package and class: foo.testimport.Foo
- in module Bar import module Foo
- In run.ceylon type: ```import { foo.testimpor``` and autocomplete with option ```{...}``` 
- Now module.ceylon is with import of Foo duplicated

**Comment by bjansen** [comment]
OK, I was able to reproduce it.

**Comment by bjansen** [comment]
This doesn't happen when `foo.testimport` is shared, so I think the package shouldn't even be proposed if it's not shared.

**Comment by bjansen** [comment]
As I suspected, this also happens in Ceylon IDE 1.2.0.

The completion tries to find a *visible* package that starts with `foo.testimpor`, and if it's not found it looks for *any* package with that name in other module repositories, even if they are not shared.

**Comment by bjansen** [comment]
@gavinking I'm thinking of two solutions here:

* import the non-shared package and do not modify modules.ceylon, this will underline the import statement and let the user know that the package is not shared
* do not add the non-shared package to the proposals, which seems to be tricky because `ModuleVersionDetails` does not store package visibility

wdyt?

**Comment by bjansen** [comment]
Fixed on master in https://github.com/ceylon/ceylon-ide-common/commit/1c082e1af5aab53833fa1cdc038ffc6dbd869de8

Queried packages won't be added to the proposal list if they come from an already imported module.

**Comment by gavinking** [comment]
> * do not add the non-shared package to the proposals, which seems to be tricky because `ModuleVersionDetails` does not store package visibility

Well a non-`shared` package in a different module is never visible to us, so should never be proposed.

**Comment by bjansen** [comment]
I don't know how to avoid proposing that: when we query the module repository, all it gives us is a list of packages, with no information on their visibility.

**Comment by gavinking** [comment]
Well I think it shouldn't do that. That's inefficient and confusing to the user. Could you open an issue against CMR?

**Comment by bjansen** [comment]
I fixed the original problem on master and 1.2.0-fixes, the module is not imported a second time.

The package visibility issue has been filed here: https://github.com/ceylon/ceylon-module-resolver/issues/133
